source/sessions/snapshot-sessions.rst

```diff
@@ -0,0 +1,339 @@
+
+Terms
+-----
+
+ClientSession
+    The driver object representing a client session and the operations that can be
+    performed on it.
+
+MongoClient
+    The root object of a driver's API. MAY be named differently in some drivers.
+
+MongoCollection
+    The driver object representing a collection and the operations that can be
+    performed on it. MAY be named differently in some drivers.
+
+MongoDatabase
+    The driver object representing a database and the operations that can be
+    performed on it. MAY be named differently in some drivers.
+
+ServerSession
+    The driver object representing a server session.
+
+Session
+    A session is an abstract concept that represents a set of sequential
+    operations executed by an application that are related in some way. This
+    specification defines how sessions are used to implement snapshot reads.
+
+Snapshot reads
+    Reads with readconcern level ``snapshot`` that occur outside of transactions on
+    both the primary and secondary nodes, including in sharded clusters.
+
+Snapshot timestamp
+    Snapshot timestamp, representing timestamp of the first read (find/aggregate/distinct operation) in the session.
+    The server creates a cursor in response to a snapshot find/aggregate command and 
+    reports ``atClusterTime`` field in the cursor response. For distinct command server adds ``atClusterTime`` field to the distinct response object. ``atClusterTime`` field represents the timestamp
+    of the read and is guaranteed to be majority committed.
+
+Specification
+=============
+
+An application requests snapshot reads by creating a ``ClientSession``
+with options that specify that snapshot reads are desired. An
+application then passes the session as an argument to methods in the
+``MongoDatabase`` and ``MongoCollection`` classes. Read operations (find/aggregate/distinct) performed against
+that session will be read from the same snapshot.
+
+High level summary of the API changes for snapshot reads
+========================================================
+
+Snapshot reads are built on top of client sessions.
+
+Applications will start a new client session for snapshot reads like
+this:
+
+.. code:: typescript
+
+    options = new SessionOptions(isSnapshot = true);
+    session = client.startSession(options);
+
+All read operations performed using this session will be read from same snapshot.
+
+If no value is provided for ``isSnapshot`` a value of false is
+implied.
+
+MongoClient changes
+===================
+
+There are no API changes to ``MongoClient`` to support snapshot reads.
+Applications indicate whether they want snapshot reads by setting the
+``isSnapshot`` field in the options passed to the ``startSession`` method.
+
+SessionOptions changes
+======================
+
+``SessionOptions`` change summary
+
+.. code:: typescript
+
+    class SessionOptions {
+        Optional<bool> isSnapshot;
+
+        // other options defined by other specs
```

Comment:
Is this to leave room for a potential `snapshotTimestamp` option in the future? Why do we need this comment?